### PR TITLE
Fix for race condition during shallow copy deletion

### PR DESCRIPTION
**Pull request by hemalathagajendran**
## Pull request checklist

Please check the type of change your PR introduces:
- [ x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- Race condition happens when shallow copy deletion and snapshot deletion happens in parallel

## What is the new behavior?

- Race condition doesn't happen when shallow copy deletion and snapshot deletion happens in parallel. Also when snapshot directory is deleted manually then also shallow copy deletion would get successful which was not happening previously.

## How risky is this change?
- [ ] Small, isolated change
- [ x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by hemalathagajendran** [comment]
Image: quay.io/hemalatha_gajendran/shallowcopy_delracecondition:latest

**Review by saurabhwani5 — DISMISSED** [review]
[issue](https://github.com/IBM/ibm-spectrum-scale-csi/issues/1105) is not getting recreated using this pr

**Review by Jainbrt — DISMISSED** [review]

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -2105,8 +2105,13 @@ func (cs *ScaleControllerServer) DeleteShallowCopyRefPath (ctx context.Context,
 	if isShallowCopyRefPathDeleted{
 		statInfo, err := conn.StatDirectory(ctx, FilesystemName, ShallowCopyRefPath)
     		if err != nil{
-      			klog.Errorf("[%s] unable to stat directory using FS [%s] at path [%s]. Error [%v]", loggerId, FilesystemName, ShallowCopyRefPath, err)
-      			return err
+			if strings.Contains(err.Error(), "EFSSG0264C") ||
+			   strings.Contains(err.Error(), "does not exist") {
+				klog.Infof("[%s] snapshot path [%s] is already deleted", loggerId, ShallowCopyRefPath)
```
Comment (deeghuge) [inline driver/csiplugin/controllerserver.go:2110]: return success from here itself

**Review by deeghuge — CHANGES_REQUESTED** [review]

File: driver/csiplugin/controllerserver.go (hunk 1 above)
Comment (hemalathagajendran) [reply driver/csiplugin/controllerserver.go:2110]: Done

**Review by deeghuge — APPROVED** [review]
LGTM

**Pull request merged**